src/components/features/memorial/memorial.js

```diff
@@ -0,0 +1,224 @@
+
+  const user = authService.getCurrentUser();
+
+  // Upvote button
+  const likesBtn = createEl("button", {
+    class: `likes upvote-btn${project.userHasVoted ? " voted" : ""}`,
+    "aria-label": "Upvote this project",
+    "aria-pressed": project.userHasVoted ? "true" : "false",
+    type: "button",
+  });
+  likesBtn.innerHTML = `
+    <span class="icon">
+      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="rgba(224,199,158,1)" width="30" height="30"><path d="M2 8.99997H5V21H2C1.44772 21 1 20.5523 1 20V9.99997C1 9.44769 1.44772 8.99997 2 8.99997ZM7.29289 7.70708L13.6934 1.30661C13.8693 1.13066 14.1479 1.11087 14.3469 1.26016L15.1995 1.8996C15.6842 2.26312 15.9026 2.88253 15.7531 3.46966L14.5998 7.99997H21C22.1046 7.99997 23 8.8954 23 9.99997V12.1043C23 12.3656 22.9488 12.6243 22.8494 12.8658L19.755 20.3807C19.6007 20.7554 19.2355 21 18.8303 21H8C7.44772 21 7 20.5523 7 20V8.41419C7 8.14897 7.10536 7.89462 7.29289 7.70708Z"></path></svg>
+    </span>
+    <span class="likes-count">${project.upvoteCount}</span>
+  `;
+  if (!user || !user.id) {
+    likesBtn.disabled = true;
+    likesBtn.title = "Login to upvote";
+    likesBtn.style.opacity = "0.6";
+    likesBtn.style.cursor = "not-allowed";
+  }
+  likesBtn.onclick = async () => {
+    if (!user || !user.id) return;
+    if (errorMsg) errorMsg.textContent = "";
+    try {
+      const res = await api.toggleUpvote(project.id);
+      if (res.success) {
+        likesBtn.classList.toggle("voted", res.data.upvoted);
+        likesBtn.querySelector(".likes-count").textContent = res.data.count;
+      } else if (errorMsg) {
+        errorMsg.textContent = res.message || "Failed to upvote project.";
+      }
+    } catch (err) {
+      if (errorMsg) errorMsg.textContent = "Error toggling upvote.";
+      console.error(err);
+    }
+  };
+
+  // Comment button
+  const commentsBtn = createEl("button", {
+    class: "comments comment-btn",
+    "aria-label": "Write a comment",
+    type: "button",
+  });
+  commentsBtn.innerHTML = `
+    <span class="icon">
+      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="rgba(224,199,158,1)" width="30" height="30"><path d="M6.45455 19L2 22.5V4C2 3.44772 2.44772 3 3 3H21C21.5523 3 22 3.44772 22 4V18C22 18.5523 21.5523 19 21 19H6.45455Z"></path></svg>
+    </span>
+    <span class="comments-count">${project.commentCount}</span>
+  `;
+  if (!user || !user.id) {
+    commentsBtn.disabled = true;
+    commentsBtn.title = "Login to comment";
+    commentsBtn.style.opacity = "0.6";
+    commentsBtn.style.cursor = "not-allowed";
+  } else {
+    commentsBtn.onclick = () => {
+      const form = document.querySelector(".comment-form textarea");
+      if (form) {
+        form.focus();
+        form.scrollIntoView({ behavior: "smooth", block: "center" });
+      }
+    };
+  }
+
+  interactions.appendChild(likesBtn);
+  interactions.appendChild(commentsBtn);
+}
+
+function renderControls(project, controls, errorMsg) {
```

Comment:
I feel like the controls for edit and delete seem a little tacked on, perhaps they need some more stylization or icons to mat the existing buttons.